specify/specify7
# Add business rule to validate Determination taxon with COType

Fixes #5111

Using business rules to check whether treeDef of a Determination Taxon is same as treeDef of the selected COType in a CO form. Save is blocked if they are not the same.

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions




#### Set up the COType field in the CO form
Ideally, COType would be set as a picklist. However, until #5098 is fixed, it is not possible to configure one for COType. Instead for this PR, COType can be set up as a QCBX. 
- Go to App Resources > Type Searches (found under Global Resources)
- Add the following line in the XML editor:
```xml
<typesearch tableid="1015"  name="CollectionObjectType" searchfield="name" displaycols="name" format="%s" uifieldformatter="" dataobjformatter="" system="true"/>
```
- Click on the Form Definition tab
- Select the correct `.views` file for the collection you are in
- Go the the Collection Object form definition
- Add the following lines inside an existing `<row>`:
```xml
<cell type="label" labelfor="111"/>
<cell type="field" id="111" name="collectionObjectType" uitype="querycbx" />
```
Your CO form should look something like this
![image](https://github.com/user-attachments/assets/a4c9bb41-f0d5-48c2-b4d2-0d297dd4927c)

#### Testing the PR
- Make sure to use a db with multiple trees in the same discipline and multiple COTypes in the same collection
- Go to the CO form
- Select a COType 
- Add a determination
- Choose a Taxon which does not belong to the COType's treeDefiniton
- [ ] Verify save is blocked
- Choose a Taxon which belongs to the COType's treeDefiniton
- [ ] Verify save is enabled
